Shimakaze-chan/Commands/Management/ManagementCommands.cs

```diff
@@ -148,6 +150,212 @@ public async Task RemoveRole(CommandContext ctx, [RemainingText] string roleStri
             await SetRole(ctx, roleString, false);
         }
 
+        [Command("purge")]
+        [Attributes.RequirePermissions(Permissions.ManageMessages)]
+        [Description("Usage: purge amount <[user]>")]
+        public async Task Purge(CommandContext ctx, [RemainingText] string suffix)
+        {
+            string[] suffixArray = suffix?.Split(" ");
+            List<ulong> usersToPurge = new List<ulong>();
+            string usersToPurgeString = "";
+            if (suffixArray.Length > 1 )
+            {
+                usersToPurge = Utils.GetIdListFromArray(ctx.Message.MentionedUsers, suffixArray.Skip(1).ToArray());
+
+
+                foreach (var id in usersToPurge)
+                {
+                    if (ctx.Guild.Members.ContainsKey(id))
+                    {
+                        usersToPurgeString += ctx.Guild.Members[id].DisplayName;
+                    }
+                    else
+                    {
+                        await CTX.RespondSanitizedAsync(ctx, $"Unable to find member with ID {id}");
+                        return;
+                    }
+                }
+            }
+
+            int purgeAmount;
+            if (!Int32.TryParse(suffixArray[0], out purgeAmount) || 
+                purgeAmount <= 0)
+            {
+                await CTX.RespondSanitizedAsync(ctx, $"{suffixArray[0]} is not a valid number between 1 and 100.");
+                return;
+            }
+            else if (purgeAmount > 100)
+            {
+                await CTX.RespondSanitizedAsync(ctx, "I can only remove up to 100 messages at a time.");
+                return;
+            }
+
+            if (usersToPurge.Count > 0)
+            {
+                DiscordMessage earliestMessage = ctx.Message;
+                List<DiscordMessage> readMessages;
+                List<DiscordMessage> messagesToDelete = new List<DiscordMessage>();
+                do
+                {
+                    readMessages = (await ctx.Channel.GetMessagesBeforeAsync(
+                        earliestMessage.Id, ShimaConsts.MaxMessageHistoryLoadCount)).ToList();
+                    if (readMessages.Count > 0)
+                    {
+                        earliestMessage = readMessages.Last();
+                    }
+                    for (int i = 0; i < readMessages.Count; i++)
+                    {
+                        if (usersToPurge.Contains(readMessages[i].Author.Id))
+                        {
+                            messagesToDelete.Add(readMessages[i]);
+                            if (messagesToDelete.Count == purgeAmount)
+                            {
+                                break;
+                            }
+                        }
+                    }
```

Comment:
This could become a foreach for readability.
```suggestion
                   foreach (var message in readMessages.Where(message => usersToPurge.Contains(message.Author.Id)))
                    {
                        messagesToDelete.Add(message);
                        if (messagesToDelete.Count == purgeAmount)
                        {
                            break;
                        }
                    }
```